railties/lib/rails/generators/rails/app/templates/config/boot.rb.tt

```diff
@@ -1,13 +1,13 @@
-<% unless options.skip_spring? -%>
-begin
+# frozen_string_literal: true
+
+ENV['BUNDLE_GEMFILE'] ||= File.expand_path('../Gemfile', __dir__)
+<% if spring_install? -%>
+
+if !defined?(Spring) && [nil, "development", "test"].include?(ENV["RAILS_ENV"])
```

Comment:
for CI these changes are not working with the default Gemfile on the board and recommendations from rails/spring. Load error handling has provided support of the CI.

So bin/spring only available on a developers machine, on production and test instances we do not have it.